posts/GSoC 2019.md

```diff
@@ -0,0 +1,64 @@
+-   missing 'E'
+-   whatever else we find
+
+Chemistry errors
+
+-   discontinuous polynomicals
+-   exceeding collision limit
+-   inconsistent thermo for explicit reverse rates
+-   negative rate coefficient (as a  [result](https://github.com/comocheng/wiki/issues/375#)  of sum of arrhenius or PLOG expressions, eg.  [here](https://github.com/Cantera/cantera-website/issues/77))
+-   excess stiffness
+-   making  [error](https://github.com/comocheng/wiki/issues/375#)  messages from CVODES easier to understand
+-   making errors from CVODES easier to catch (raise a python Error)
+-   dead-end pathways
+## How Can I Achieve It
+
+The debugging tool PyCanChemAndYouCan2 is a web-based application developed in Django 2.2.  Users can upload original files and download revised files from the website, and the application contains  ACE code editor to allow users edit their files on the website. For Debugging script, Cantera has `ck2cti.py` file to convert different file format, so the debugging part will be test different defective mechanism files and update `ck2cti.py` file. The ideal condition is that the application can help users find the errors and sugget the fixes or fix errors automatically. However, it might be a long-term goal, so we make some minimum goal for this GSoC project.
+Here are the basic ideas about the project:
+ -   a script that takes in an input file and returns an input file with added comments, which describe detected errors.
+-   a web site that makes it easy to interact with that script (which runs on the web server), and renders the comments in an attractive way
+
+The idea is that the script can be used, developed, tested, as a stand-alone thing without the website.  
+The website just makes things look prettier. eg. a custom "syntax highlighting" mode that detects comments describing certain problems and renders them in red, or whatever.
+
+Here is the workflow of the application:
+
+```
+A[upload files] --> B[list files] --> C{choose one and execute by ck2cte.py}  --> D{show input file with added comments on ACE editor}
+--> E[edit file] --> F[download edited file] 
+```
+## How Much Have I Got
+I have got the upload and download functions and merged ACE code editor into the web-application, the current challenge is how to display the files on the ACE code editor after the file has been executed by `ck2cti.py`.
+I am looking for a way to execute external python script in a Django-developed website, or how can I execute the python file using Javascript language. I am refering [django-ace]([https://github.com/django-ace/django-ace](https://github.com/django-ace/django-ace)), [RMGwebsite]([https://github.com/ReactionMechanismGenerator/RMG-website](https://github.com/ReactionMechanismGenerator/RMG-website)).
+
+My project code was posted on https://github.com/comocheng/ChemCheck, and I will update it every Sunday.
+
+Please leave your comments here if you have any suggestions for the project.
```

Comment:
We don't have comments enabled on the website :-) You could suggest that people post on Google Groups, I will (or you can) create a topic once this blog post goes live to announce the project.